packages/openid4vc-holder/src/OpenId4VcHolderApi.ts

```diff
@@ -0,0 +1,155 @@
+import type {
+  ResolvedCredentialOffer,
+  AuthCodeFlowOptions,
+  AcceptCredentialOfferOptions,
+  ResolvedAuthorizationRequest,
+} from './OpenId4VcHolderServiceOptions'
+import type { VerificationMethod, W3cCredentialRecord } from '@aries-framework/core'
+import type { ClientMetadataOpts, VerifiedAuthorizationRequest } from '@sphereon/did-auth-siop'
+import type { CredentialOfferPayloadV1_0_11 } from '@sphereon/oid4vci-common'
+
+import { injectable, AgentContext } from '@aries-framework/core'
+
+import { OpenId4VcHolderService } from './issuance/OpenId4VciHolderService'
+import { OpenId4VpHolderService } from './presentations/OpenId4VpHolderService'
+
+/**
+ * @public
+ */
+@injectable()
+export class OpenId4VcHolderApi {
+  private agentContext: AgentContext
+  private openId4VcHolderService: OpenId4VcHolderService
+  private openId4VpHolderService: OpenId4VpHolderService
+
+  public constructor(
+    agentContext: AgentContext,
+    openId4VcHolderService: OpenId4VcHolderService,
+    openId4VpHolderService: OpenId4VpHolderService
+  ) {
+    this.agentContext = agentContext
+    this.openId4VcHolderService = openId4VcHolderService
+    this.openId4VpHolderService = openId4VpHolderService
+  }
+
+  public async createRequest(options: {
+    verificationMethod: VerificationMethod
+    redirect_url: string
+    clientMetadata?: ClientMetadataOpts
+    issuer?: string
+  }) {
+    const relyingParty = await this.openId4VpHolderService.getRelyingParty(this.agentContext, options)
+
+    // TODO: generate nonce, state, correlationId
+    const nonce = 'qBrR7mqnY3Qr49dAZycPF8FzgE83m6H0c2l0bzP4xSg'
+    const state = 'b32f0087fc9816eb813fd11f'
+    const correlationId = '1'
```

Comment:
This should be fixed before merging, correct?

Reply:
Yes